README.md

```diff
@@ -1,175 +1,529 @@
+Tell us what you want to see by opening an [issue](https://github.com/crystaldba/postgres-mcp/issues) or a [pull request](https://github.com/crystaldba/postgres-mcp/pulls).
+You can also contact us on [Discord](https://discord.gg/4BEHC7ZM).
+
+## Technical Notes
+
+This section includes a high-level overview technical considerations that influenced the design of Postgres Pro.
+
+### Index Tuning
+
+Developers know that missing indexes are one of the most common causes of database performance issues.
+Indexes provide access methods that allow Postgres to quickly locate data that is required to execute a query.
+When tables are small, indexes make little difference, but at the size of the data grows, the difference in algorithmic complexity between a table scan and an index lookup becomes significant (typically *O*(*n*) vs *O*(*log* *n*), potentially more if joins on multiple tables are involved).
+
+Generating suggested indexes in Postgres Pro proceeds in several stages:
+
+1. *Identify SQL queries in need of tuning*.
+    If you know you are having a problem with a specific SQL query you can provide it.
+    Postgres Pro can also analyze the workload to identify index tuning targets.
+    To do this, it relies on the `pg_stat_statements` extension, which records the runtime and resource consumption of each query.
+
+    A query is a candidate for index tuning if it is a top resource consumer, either on a per-execution basis or in aggregate.
+    At present, we use execution time as a proxy for cumulative resource consumption, but it may also make sense to look at specifics resources, e.g., the number of blocks accessed or the number of blocks read from disk.
+
+    Sophisticated index tuning systems use "workload compression" to produce a representative subset of queries that reflects the characteristics of the workload as a whole, reducing the problem for downstream algorithms.
+    Postgres Pro performs a limited form of workload compression by normalizing queries so that those generated from the same template appear as one.
+    It weights each query equally, a simplification that works when the benefits to indexing are large.
+
+2. *Generate candidate indexes*
+    Once we have a list of SQL queries that we want to improve through indexing, we generate a list of indexes that we might want to add.
+    To do this, we parse the SQL and identify any columns used in filters, joins, grouping, or sorting.
+
+    To generate all possible indexes we need to consider combinations of these columns, because Postgres supports [multicolumn indexes](https://www.postgresql.org/docs/current/indexes-multicolumn.html).
+    In the present implementation, we include only one permutation of each possible multicolumn index, which is selected at random.
+    We make this simplification to reduce the search space because permutations often have equivalent performance.
+    However we hope to improve in this area.
```

Comment:
"However we hope to improve in this area." → missing comma: "However, we hope to improve in this area."